### ISSUE TITLE
Hooks?

### ISSUE THREAD

**Issue by rektide** [hybridsjs/hybrids#45]
Hello.

Any thoughts on how we might get a [`useState`](https://reactjs.org/docs/hooks-reference.html#useeffect) like capability in Hybrids?

Or other capabilities in [React Hooks](https://reactjs.org/docs/hooks-reference.html)?

I feel like Hybrids has a more explicit idea of state- that the component itself keeps getting its properties passed to it, via the arguments each render() call gets. This may perhaps make hooks unnecessary, but I'm not sure. What are your thoughts? Perhaps some additional documentation could be added to help onboard users used to React Hooks, as they come to Hybrids' functional component model?

**Comment by smalluban** [comment]
In simple words, it is just not fit into the hybrids architecture. Hooks are the answer for creating a state inside of the procedural and sequential called function. The hybrids use an object with properties for creating a shape of the component, so what is a `useState` for React, property factory or simple definition of the property is for hybrids.

Let's take a simple example from React docs:

```javascript
function Counter({initialCount}) {
  const [count, setCount] = useState(initialCount);
  return (
    <>
      Count: {count}
      <button onClick={() => setCount(initialCount)}>Reset</button>
      <button onClick={() => setCount(prevCount => prevCount + 1)}>+</button>
      <button onClick={() => setCount(prevCount => prevCount - 1)}>-</button>
    </>
  );
}
```
You would write the same in hybrids like this:
```javascript
const Counter = {
  initialCount: 0,
  count: { connect: (host, key) => { host[key] = host.initialCount; } },
  render: ({ count }) => html`
    Count: ${count}
    <button onclick=${host => { host.count = host.initialCunt; }}>Reset</button>
    <button onclick=${host => { host.count += 1; }}>+</button>
    <button onclick=${host => { host.count -= 1; }}>-</button>
  `,
};
```

As you can see, there is simply no place for the hooks - you set declaratively your inputs and in a separate function, you render things. That architecture does not require special care when testing render function, as hooks don't work outside of the react ecosystem. Properties defined by hybrids don't have to be placed in special order - you don't have to depend on new eslint rules to protect you, etc..

To sum up, I have no future plans to implement a similar idea to hooks in hybrids. About your idea of extending docs - I am not sure if there are so many react hooks users out there, and I think the architecture is so different, that I would not compare it to hooks to avoid creating a misleading picture of the library.
